### PR TITLE
New Integration with Camlytics

### PR DESCRIPTION
**Pull request by ayman-m**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
A new integration with Camlytics, a surveillance analysis system.

## Minimum version of Cortex XSOAR
- [x ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ x] No

## Must have
- [ ] Tests
- [x ] Documentation

### CODE REVIEW

**Comment by merit-maita** [comment]
Hello @ayman-m
can you please fell the [registration form](https://forms.gle/XDfxU4E61ZwEESSMA) so we can go ahead with the review?
thanks

File: Packs/Camlytics/Integrations/Camlytics/Camlytics.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+        if command == 'test-module':
+            test_module(client)
+        elif command in commands:
+            return_results(commands[command](client, args))
+        else:
+            raise NotImplementedError(f'{command} command is not implemented.')
+
+    except Exception as e:
+        return_error(str(e))
```
Comment (merit-maita) [inline Packs/Camlytics/Integrations/Camlytics/Camlytics.py:145]:
```suggestion
        return_error(f'Failed to execute {command} command.\nError:\n{str(e)}')
```

File: Packs/Camlytics/Integrations/Camlytics/Camlytics.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    demisto.debug(f'Command being called is {command}')
+
+    try:
+        requests.packages.urllib3.disable_warnings()
+        client: Client = Client(urljoin(url, ''), verify_certificate, proxy, headers=headers)
+        commands = {
+            'camlytics-get-channels': channels_command,
+            'camlytics-get-events-totals-by-rule': eventstotalsrule_command,
+            'camlytics-get-events-totals-by-type': eventstotalstype_command,
+            'camlytics-get-events': events_command,
```
Comment (merit-maita) [inline Packs/Camlytics/Integrations/Camlytics/Camlytics.py:134]:
```suggestion
            'camlytics-get-channels': get_channels_command,
            'camlytics-get-events-totals-by-rule': get_events_totals_by_rule_command,
            'camlytics-get-events-totals-by-type': get_events_totals_by_type_command,
            'camlytics-get-events': get_events_command,
```

File: Packs/Camlytics/Integrations/Camlytics/Camlytics.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        params = assign_params(channelId=channelid, sinceId=sinceid, sinceTime=sincetime,
+                               untilId=untilid, untilTime=untiltime, limit=limit, order=order, timeout=timeout)
+        headers = self._headers
+
+        response = self._http_request('GET', 'events', params=params, headers=headers)
+
+        return response
+
+
+def channels_command(client: Client, args: Dict[str, Any]) -> CommandResults:
```
Comment (merit-maita) [inline Packs/Camlytics/Integrations/Camlytics/Camlytics.py:42]:
```suggestion
def get_channels_command(client: Client, args: Dict[str, Any]) -> CommandResults:
```

File: Packs/Camlytics/Integrations/Camlytics/Camlytics.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        outputs_prefix='Camlytics.Channels',
+        outputs_key_field='channel_id',
+        outputs=response,
+        raw_response=response
+    )
+
+    return command_results
+
+
+def eventstotalsrule_command(client: Client, args: Dict[str, Any]) -> CommandResults:
```
Comment (merit-maita) [inline Packs/Camlytics/Integrations/Camlytics/Camlytics.py:55]:
```suggestion
def get_events_totals_by_rule_command(client: Client, args: Dict[str, Any]) -> CommandResults:
```

File: Packs/Camlytics/Integrations/Camlytics/Camlytics.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        outputs_prefix='Camlytics.EventsTotalsRule',
+        outputs_key_field='',
+        outputs=response,
+        raw_response=response
+    )
+
+    return command_results
+
+
+def eventstotalstype_command(client: Client, args: Dict[str, Any]) -> CommandResults:
```
Comment (merit-maita) [inline Packs/Camlytics/Integrations/Camlytics/Camlytics.py:72]:
```suggestion
def get_events_totals_by_type_command(client: Client, args: Dict[str, Any]) -> CommandResults:
```

File: Packs/Camlytics/Integrations/Camlytics/Camlytics.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+        outputs_prefix='Camlytics.EventsTotalsType',
+        outputs_key_field='',
+        outputs=response,
+        raw_response=response
+    )
+
+    return command_results
+
+
+def events_command(client: Client, args: Dict[str, Any]) -> CommandResults:
```
Comment (merit-maita) [inline Packs/Camlytics/Integrations/Camlytics/Camlytics.py:89]:
```suggestion
def get_events_command(client: Client, args: Dict[str, Any]) -> CommandResults:
```

**Review by merit-maita — CHANGES_REQUESTED** [review]
Thank you for your great effort!

**Comment by ayman-m** [comment]
> Thank you for your great effort!

Thank you, I pushed another commit now

**Comment by merit-maita** [comment]
@ayman-m Thank you for the hard work!
i'd like to schedule a demo with you, please let me know when it suits you.

**Review by merit-maita — APPROVED** [review]
Thank you Ayman, very good job
one last step before merging,  the validation check fails because of a lint error with mypy.
that was solved. please pull from master and push again.

**Comment by ayman-m** [comment]
> Thank you Ayman, very good job one last step before merging, the validation check fails because of a lint error with mypy. that was solved. please pull from master and push again.

Thanks Merit for your time during the session, I have pulled from the master and pushed again, the branch is now in sync

**Comment by merit-maita** [comment]
> > Thank you Ayman, very good job one last step before merging, the validation check fails because of a lint error with mypy. that was solved. please pull from master and push again.
> 
> Thanks Merit for your time during the session, I have pulled from the master and pushed again, the branch is now in sync

you're most welcome!
i see that lint fails overa different error, please take a look and fix it.

**Comment by ayman-m** [comment]
> > > Thank you Ayman, very good job one last step before merging, the validation check fails because of a lint error with mypy. that was solved. please pull from master and push again.
> > 
> > 
> > Thanks Merit for your time during the session, I have pulled from the master and pushed again, the branch is now in sync
> 
> you're most welcome! i see that lint fails overa different error, please take a look and fix it.

Hi @merit-maita , I fixed it

**Pull request merged**